CONTRIBUTING.md

```diff
@@ -110,6 +110,13 @@ must include:
    These are located in `/test/end-to-end-tests`. Ideally, you would also
    include tests for edge and error cases.
 
+Unit tests are expected even when the feature is in labs. It's good practice
```

Comment:
I think if we're saying some code can escape the threshold of typical quality, we need a more powerful distinction for marking it as "experimental" than just labs. A lot of labs code these days really is intended for eventual production, and skipping tests during development is problematic because adding tests later usually doesn't work - the code won't be very testable and will need reworking.